lib.nix

```diff
@@ -37,7 +66,10 @@ rec
           {
             inherit (p) name version;
             sha256 = p.checksum;
-          }
+          } // (if builtins.hasAttr "registry" p
+            then { inherit (registries.${p.registry}) index; }
+            else { }
+          )
```

Comment:
```suggestion
          } // (lib.optionalAttrs (builtins.hasAttr "registry" p) {
              inherit (registries.${p.registry}) index;
            }
          )
```

otherwise LGTM